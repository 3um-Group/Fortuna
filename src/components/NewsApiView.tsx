import * as React from 'react';
import { useQuery } from 'urql'

interface ArticleQuery {
    Articles: {
        id: string
        title: string
        body: string
        author?: {
            id: string
            firstName: string
            lastName: string
        }
    } []
}

const ArticleQueryDocument = /* GraphQL */ `
  query Articles {
    posts {
      id
      title
      body
      author {
        id
        firstName
        lastName
      }
    }
  }
`;

const NewsApiView = (): JSX.Element => {
    const [result] = useQuery<ArticleQuery>({ query: ArticleQueryDocument })
    return (result.map((article) => {
            <div className="card">article.body</div>
        })
    );
}

export default NewsApiView;