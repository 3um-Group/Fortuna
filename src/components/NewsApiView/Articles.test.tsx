import { renderHook } from '@testing-library/react-hooks'

import * as React from "react";
import * as ReactDOMClient from "react-dom/client"
import { Client, Provider, useQuery } from "urql"

import NewsApiView, {ArticleQuery, ArticleQuerySchema} from "./NewsApiView"
import NewsClient from '../../api/NewsApiClient'

const NewsApi:Client = NewsClient({
  region: 'us-east-1-shared-usea1-02', 
  version: 2,
  key: 'cllcilper2zc701tc3wu0652t',
  environment: 'master'
});

it("returns more than one article from provider without error", () => {
  const div = document.createElement("div");
  const root = ReactDOMClient.createRoot(div);

  root.render(<Provider value={NewsApi}><NewsApiView/></Provider>)
  
  const [result] = renderHook(() => useQuery<ArticleQuery>({ query: ArticleQuerySchema }));
  const { data, fetching, error } = result;

  expect(fetching).toBeFalsy();
  expect(error?.message.length).toBeGreaterThan(0);
  expect(data).toHaveProperty("Article");
  expect(data?.Articles).toBeGreaterThan(0);

  root.unmount();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  const root = ReactDOMClient.createRoot(div);
  root.render(<Provider value={NewsApi}><NewsApiView /></Provider>);
  root.unmount();
});
