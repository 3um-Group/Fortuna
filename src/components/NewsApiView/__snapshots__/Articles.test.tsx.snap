// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 2`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 3`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 4`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 5`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 6`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 7`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 8`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 9`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 10`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 11`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 12`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 13`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 14`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 15`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 16`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 17`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 18`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 19`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 20`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;

exports[`renders without crashing 21`] = `
<DocumentFragment>
  <p>
    Oh no... [GraphQL] [line: 1] field 'body' is not defined in 'Post'
  </p>
</DocumentFragment>
`;
